src/uaclient/uasession.cpp

```diff
@@ -73,6 +76,30 @@ UaStatus UaSession::connect(
     return OpcUa_Good;
 }
 
+/**
+ * TODO: serviceSettings,
+ * TODO: deleteSubscriptions
+ */
+UaStatus UaSession::disconnect(
+        ServiceSettings &       serviceSettings,
+        OpcUa_Boolean           deleteSubscriptions
+    )
+{
+    std::lock_guard<decltype(m_accessMutex)> lock (m_accessMutex);
+    if (! m_client)
+    {
+        LOG(Log::WRN) << "Can't disconnect because not connected.";
+        return OpcUa_BadInvalidState;  // can't disconnect if not connected...
+    }
+    UaStatus status = UA_Client_disconnect(m_client);
+    if (m_client)
+    {
+        UA_Client_delete(m_client);
+        m_client = 0;
+    }
```

Comment:
Would it be useful to check if  indeed m_client doesn't exist after deleting it before returning?
Maybe throwing an exception otherwise?

Reply:
I'm not sure what's on your mind - UA_Client_delete doesn't return any status. Could you elaborate?

Reply:
Exactly! Thats why I was wondering if a further check would be useful after it. But looking at it again it would not.